### PR TITLE
Update Docker Images of vala-lint

### PR DESCRIPTION
**Pull request by kgrubb**
This pr should set everything up to enable the vala-lint image to be auto-updated every time there's a push/merge to master. This should hopefully satisfy #96!

**_Note:_** This requires an administrator of the vala-lint project to set up the docker login secrets. After doing so, any merge to master will automatically kick off a new github workflow that will build and publish an updated image.

### CODE REVIEW

**Comment by btkostner** [comment]
Everything looks like it would work, but we recently made a couple of changes to the org, so lets make this a bit better.

1) Let's use the `valalang` dockerhub organization. The full image should be `valalang/lint:latest`
2) Let's not use the elementary image as a base. We can make this smaller (and therefor faster for CI) and more future proof if we use a debian, or ubuntu base.

**Comment by kgrubb** [comment]
@btkostner I just swapped to debian:stable-slim, which cuts down the build size pretty significantly, but i did notice some of the dependencies are a bit older (valac v0.42.5-1). Everything builds fine though.

**Comment by kgrubb** [comment]
Size differences!

elementary-based image (`elementary/docker:juno-unstable`):
Screenshot from 2019-09-05 11-31-11

debian-based image (`debian:stable-slim`):
Screenshot from 2019-09-05 11-33-32

**Comment by kgrubb** [comment]
So, this is probably _too_ crazy, but I just noticed that alpine has [vala in apk](https://pkgs.alpinelinux.org/packages?name=*vala*&branch=edge). I might see if i can get vala-lint running on an alpine container, because that'd make this image super duper tiny. Plus it'd be kinda fun :laughing:

**Comment by btkostner** [comment]
I'm down to `260MB` right now

**Comment by btkostner** [comment]
For some reason, it's not letting me do a git push, so here is the Dockerfile I was able to get working:

```
# elementary docker image for vala-lint

FROM debian:stable-slim

ENV DEBIAN_FRONTEND=noninteractive

RUN mkdir -p /opt/vala-lint-portable

COPY . /opt/vala-lint

RUN apt update \
  && apt install -y libvala-dev valac meson

RUN cd /opt/vala-lint \
  && meson build --prefix=/usr \
  && cd build \
  && ninja \
  && DESTDIR=/opt/vala-lint-portable ninja install

FROM debian:stable-slim

ENV DEBIAN_FRONTEND=noninteractive

COPY --from=0 /opt/vala-lint-portable /

RUN apt update \
  && apt install -y libvala-dev gio-2.0

RUN mkdir -p /app
VOLUME /app
WORKDIR /app

CMD ["/usr/bin/io.elementary.vala-lint", "**/*.vala"]
```

File: README.md [hunk 1]
```diff
@@ -45,4 +45,4 @@ if(...) { // vala-lint=space-before-paren, line-length
 ### Docker and Continuous Integration
 Vala-Lint is primarily intended to be used in Continuous Integration (CI). It's available in a convenient, always up-to-date Docker container `elementary/docker:vala-lint` hosted on Docker Hub.
```
Comment (danrabbit) [inline README.md]: one more :)

```suggestion
Vala-Lint is primarily intended to be used in Continuous Integration (CI). It's available in a convenient, always up-to-date Docker container `valalang/lint:latest` hosted on Docker Hub.
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    container:
+      image: elementary/docker:unstable
```
Comment (btkostner) [inline .github/workflows/publish.yml]: I'm pretty sure this is useless

**Comment by btkostner** [comment]
`docker run -v "$PWD":/app valalang/lint`

**Pull request merged**